### PR TITLE
Feature/weather function

### PR DESCRIPTION
**Pull request by Reptilian6163**
Added function handling for the GPT Assistant.

The assistant handles a single message-response interaction for now.

Function does not return actual weather data.

### CODE REVIEW

File: server/src/services/assistantService.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      } else if (message.role === 'function') {
+        console.log(`Function: ${message.content[0].text.value}`);
+      }
+    });
+
+  } catch (error) {
+    console.error("Error displaying messages:", error);
+    throw error;
+  }
+}
```
Comment (HlexNC) [inline server/src/services/assistantService.js:161]: What is the purpose of this function, can't this be tested with swagger and postman?

File: server/src/controllers/assistantController.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    #swagger.responses[500] = { description: 'Server error' }
+  */
```
Comment (HlexNC) [inline server/src/controllers/assistantController.js:37]: Please add a swagger example that worked when you were testing. And (if possible) an output sample. For Lex/me.

File: server/src/services/assistantService.js (hunk 1 above)
Comment (Reptilian6163) [reply server/src/services/assistantService.js:161]: This simply log the messages of user and assistant. I didn't understand how to have the Swagger thing add said messages as a response in Swagger, and I wanted a quick way to test if everything was working as intended.

File: server/src/controllers/assistantController.js (hunk 2 above)
Comment (Reptilian6163) [reply server/src/controllers/assistantController.js:37]: So, add an output and an example for the post request?

File: server/src/services/assistantService.js (hunk 1 above)
Comment (HlexNC) [reply server/src/services/assistantService.js:161]: > This simply log the messages of user and assistant. I didn't understand how to have the Swagger thing add said messages as a response in Swagger, and I wanted a quick way to test if everything was working as intended.

Swagger simply logs the output, the `res` can be anything you define.

File: server/src/controllers/assistantController.js (hunk 2 above)
Comment (Reptilian6163) [reply server/src/controllers/assistantController.js:37]: Okay, I see. I'll try implementing it by tonight.

File: server/src/services/assistantService.js (hunk 1 above)
Comment (Reptilian6163) [reply server/src/services/assistantService.js:161]: I don't entirely understand, but I'll try to have Swagger take user-defined message and outputs.

**Pull request merged**